script/tool/lib/src/common/plugin_command.dart

```diff
@@ -15,6 +15,19 @@ import 'core.dart';
 import 'git_version_finder.dart';
 import 'process_runner.dart';
 
+/// An entry in pacakge enumeration for APIs that need to include extra
+/// data about the entry.
+class PackageEnumerationEntry {
```

Comment:
It's a bit odd that this abstraction name includes the context that it is used.  Couldn't this just be a `Package`?

Reply:
This is public throughout the entire codebase, which would make calling it just "Package" confusing; most references to packages in the tool are not enumerations, just individual entries.

The reason the context is part of the name is that the extra data is specific to enumeration—it's what it is excluded *from*. It's indicating that even though the package is in the enumeration, it shouldn't really be handled as part of it. (Originally I wanted to separate excluded packages out, since having them in the list is inherently somewhat confusing, but that doesn't allow for solving the sharding problem—their place in the list must be preserved.)